### PR TITLE
Fix when the last common block before a fork is added after the fork

### PR DESCRIPTION
**Pull request by gorillainduction**
[Fixes #153315211]

### CODE REVIEW

File: apps/aecore/test/aec_chain_state_tests.erl [hunk 1]
```diff
@@ -662,6 +663,25 @@ fork_common_headers(EasyChain, TopHashEasy, HardChain, TopHashHard) ->
...
+
+    State1 = write_blocks_to_chain(butlast(CommonChain), InitState),
+    {ok, State2} = insert_block(lists:last(EasyChain), State1),
+    {ok, State3} = insert_block(lists:last(HardChain), State2),
+
+    %% The last block to enter is the last common node.
+    ?assertMatch({ok, _}, insert_block(lists:last(CommonChain), State3)),
+    ok.
+
+butlast([_]) -> [];
```
Comment (hanssv) [inline apps/aecore/test/aec_chain_state_tests.erl]: My god... This is a standard function (after I nagged the OTP team for over a year...) :-)
`lists:droplast/1`

File: apps/aecore/test/aec_chain_state_tests.erl [hunk 2]
```diff
@@ -662,6 +663,25 @@ fork_common_headers(EasyChain, TopHashEasy, HardChain, TopHashHard) ->
...
+
+    %% Add the chain with the fork node as the last entry.
+    InitState = new_state(),
+
+    State1 = write_blocks_to_chain(butlast(CommonChain), InitState),
+    {ok, State2} = insert_block(lists:last(EasyChain), State1),
+    {ok, State3} = insert_block(lists:last(HardChain), State2),
+
+    %% The last block to enter is the last common node.
+    ?assertMatch({ok, _}, insert_block(lists:last(CommonChain), State3)),
```
Comment (lucafavatella) [inline apps/aecore/test/aec_chain_state_tests.erl]: I was expecting checking new top here - though I am still getting understanding of PR.

File: apps/aecore/test/aec_chain_state_tests.erl (hunk 1 above)
Comment (gorillainduction) [reply apps/aecore/test/aec_chain_state_tests.erl]: I will note this, and totally ignore it ;-)

File: apps/aecore/test/aec_chain_state_tests.erl (hunk 2 above)
Comment (gorillainduction) [reply apps/aecore/test/aec_chain_state_tests.erl]: Agreed. The symptom before was to crash, though.

I'll add the test.

Comment (lucafavatella) [reply apps/aecore/test/aec_chain_state_tests.erl]: If I keep test but remove code change I see the badmatch - thanks.

Comment (lucafavatella) [reply apps/aecore/test/aec_chain_state_tests.erl]: I saw you added the checks in force-push - thanks. You have been faster than me!

**Comment by lucafavatella** [comment]
I see two approvals. I see the tests and packing job on Linux passed.
I plan to wait for a Linux Dialyzer job then merge.

File: apps/aecore/test/aec_chain_state_tests.erl (hunk 2 above)
Comment (lucafavatella) [reply apps/aecore/test/aec_chain_state_tests.erl]: I also tested locally with additional checks on top (header and block) at the several intermediate stages, all ok.

**Comment by lucafavatella** [comment]
I see CI dialyzer job passed too. And I had run tests locally on OSX.

**Pull request merged**